File: Api-SistemaMercearia/Api-SistemaMercearia/Models/Products/Produto.cs
namespace Api_SistemaMercearia.Models.Products
{
    public class Produto
    {
        public int Id { get; set; }
        public int Codigo { get; set; }
        public string Descricao { get; set; }
        public string Marca { get; set; }
        public double Valor { get; set; }
        public int Estoque { get; set; }
        public IEnumerable<ProdutoVenda>? Vendas { get; set; }

        public Produto(string descricao, string marca, double valor, int estoque)
        {
            Descricao = descricao;
            Marca = marca;
            Valor = valor;
            Estoque = estoque;
        }
    }
}

File: Api-SistemaMercearia/SPA-App/Sistema-Mercearia/src/app/Models/Produto.ts
export class Produto {
    id?: number;
    codigo?: number;
    descricao?: string;
    marca?: string;
    valor?: number;
    estoque?: number

    constructor(id ?: number,codigo ?: number, descricao ?: string, marca ?: string, valor ?: number, estoque ?: number) {
       this.id = id;
       this.codigo = codigo;
       this.descricao = descricao;
       this.marca = marca;
       this.valor = valor;
       this.estoque = estoque;
    }
}
